internal/wsutil/wsutil.go

```diff
@@ -109,48 +81,64 @@ func WebsocketSendStream(conn MessageWriter, r io.Reader, bufferSize int) chan b
 func WebsocketRecvStream(w io.Writer, conn MessageReader) chan bool {
 	ch := make(chan bool)
 
-	go func(w io.Writer, conn MessageReader) {
-		for {
-			mt, r, err := conn.NextReader()
-			if mt == websocket.CloseMessage {
-				logger.Debugf("Got close message for reader")
-				break
-			}
+	go func() {
+		recvLoop(w, conn)
+		close(ch)
+	}()
 
-			if mt == websocket.TextMessage {
-				logger.Debugf("Got message barrier")
-				break
-			}
+	return ch
+}
 
-			if err != nil {
-				logger.Debugf("Got error getting next reader %s", err)
-				break
-			}
+func recvLoop(w io.Writer, conn MessageReader) {
+	buf := make([]byte, 32*1024) // only allocate once per websocket, not once per loop
+
+	for {
+		mt, r, err := conn.NextReader()
+		if err != nil {
+			logger.Debugf("Cannot get next reader: %v", err)
```

Comment:
Not really required for now but in the future I'd probably make recvLoop return an error, that way all this error logging can be handled in the caller.